catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/converter/type/RegistryObjectTypeConverter.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.schemabindings.converter.type;
+
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.CLASSIFICATION_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.DESCRIPTION_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.EXTERNAL_IDENTIFIER_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.HOME_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.ID_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.LID_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.NAME_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.OBJECT_TYPE_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.RIM_FACTORY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.SLOT;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.STATUS_KEY;
+import static org.codice.ddf.registry.schemabindings.EbrimConstants.VERSION_INFO_KEY;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.registry.schemabindings.helper.InternationalStringTypeHelper;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ClassificationType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExternalIdentifierType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.InternationalStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.VersionInfoType;
+
+public class RegistryObjectTypeConverter {
+
+    protected static final InternationalStringTypeHelper INTERNATIONAL_STRING_TYPE_HELPER =
+            new InternationalStringTypeHelper();
+
+    public RegistryObjectTypeConverter() {
+    }
+
+    /**
+     * This method creates an RegistryObjectType from the values in the provided map.
+     * The following keys are expected in the provided map (Taken from EbrimConstants):
+     * <p>
+     * CLASSIFICATION_KEY = "Classification";
```

Comment:
Once the constants have been moved to this class, should we just have a single statement that says something like `Refer to the list of {@code _KEYS} constants for the list expected map keys` or something.